src/vs/editor/browser/view/viewLayer.ts

```diff
@@ -605,4 +607,11 @@ class ViewLayerRenderer<T extends IVisibleLine> {
 			}
 		}
 	}
+
+	private _lineHeightForLineNumber(lineNumber: number): number {
+		if (this._viewportData.specialLineHeights.has(lineNumber)) {
+			return this._viewportData.specialLineHeights.get(lineNumber)!;
+		}
+		return this._viewportData.lineHeight;
+	}
```

Comment:
Try to feed in a getter for this data via a new ctor field if necessary. Please don't add the line heights to the viewport data.